### DISCUSSION TITLE
Broken links with blog or wiki + caching problems

### DISCUSSION THREAD

**Discussion by nibreh** [datenstrom/community#12] [Report a bug] [answered]
Hello Mark - Always playing with YellowCMS, it's a great app, simple and powerfull. I will install for friends.

I noted some problems :
- EDIT : When Blog or Wiki are defined as home page (config.ini >> contentHomeDir), some links are broken in EDIT MODE (pagination, recent changes, etc)
- When making changes in files with a text editor or also sometimes in Edit mode, always must empty the cache of the browser to take effect - testing with meta tags not resolve the problem.

bad english, I know ;)

**Comment by nibreh** [comment]
Sorry, for blog or wiki as home page, it's OK.

It's just when you are in EDIT MODE, pagination link is broken for BLOG, so you can't go previous page to edit old article (only with URL)

**Comment by markseuffert** [comment]
About the first issue with the blog, I was not able to reproduce it. How did you name the top-level content folders when this happened, do you have an example installation with broken links?

About the second issue with emptying cache. Yes, this happens when you make structural changes... such as snippets, templates and folders. Sorry about that. It should never happen when you make content changes... such as editing and creating web pages.

Thanks a lot for reporting, any additional information is appreciated!

**Comment by markseuffert** [comment]
Now I can reproduce the blog editing problem. I'll create a new issue to sort this out as soon as possible.
Good that you found it! :+1:

**Comment by nibreh** [comment]
Good ! Take a look at the Wiki too, recent change also broken in EDIT MODE when wiki is home page (If I'm not wrong)

By the way, now after a fresh install (yellow-master), on localhost with xampp server, I give the index folder and files for http://localhost/yellow, haha. Problem with lampp and mod rewrite ? 

It's fine with my webhosting legtux.org

**Comment by markseuffert** [comment]
Yes, the CMS requires mod_rewrite. Can you enable it on your server?

**Comment by nibreh** [comment]
Don't know why or what I've done, but it's working now ;) - Thanks

**Comment by markseuffert** [comment, marked as answer]
The latest version should fix the broken links. Let me know if it works, thanks!

**Comment by nibreh** [comment]
Yep it's working ! good job;) thanks

**Comment by markseuffert** [comment]
Is caching still a problem? If you have more information I would like to investigate it. 

For example the following should work without emptying the browser cache: Adding a page to the blog/wiki, changing text of a page, deleting a page. At the moment the blog/wiki does not set an expire time, so a visitor is not forced to reload a page every time. Let me know what's not working as expected and we can try to fix it.

**Comment by nibreh** [comment]
The problem is, like you say, when we add some changes in snippets, templates and folders files. 

For example, adding fancybox scripts or something like that in header.php >> visitor must empty the cache to take effect when he returns. (Personally, my cache is automatically empty when I close Firefox, it's a choice and not by default in parameters)

But I am satisfied right now, I don't add many changes, and it's better to published yellow once it's ready to work.

**Comment by markseuffert** [comment]
Sorry, I don't know a good solution for site-wide changes. Right now I reload a page or empty the browser cache. Same problem here. One thing I like to do in the future is moving CSS files into the system folder. Makes it more obvious that style changes are like other system changes.

Thanks again for your feedback!

**Comment by markseuffert** [comment]
Related issue datenstrom/yellow#47
